examples/images/cifar10/train_cifar10.py

```diff
@@ -65,6 +67,12 @@ def train(argv):
         FLAGS.save_step,
     )
 
+    if FLAGS.parallel and world_size > 1:
+        # When using `DistributedDataParallel`, we need to divide the batch
+        # size ourselves based on the total number of GPUs of the current node.
+        FLAGS.batch_size = int(FLAGS.batch_size / world_size)
```

Comment:
I don't like overwriting the FLAGS's batch size. Can you rename it 'batch_size_per_gpu' or just 'batch_size? and update line 91?